scripts/postinstall.ts

```diff
@@ -120,33 +120,45 @@ async function downloadDependencies(items: DependencyWithContext[]): Promise<voi
     for (const dependent of reverseDependencies[name]) {
       if (!running.has(dependent)) {
         if (forwardDependencies[dependent].every(d => done.has(d))) {
-          promises.push(process(dependent));
+          promises[dependent] = process(dependent);
         }
       }
     }
   }
 
   for (const item of items.filter(d => (d.dependency.dependencies?.(d.context) ?? []).length === 0)) {
-    promises.push(process(specialize(item)));
+    promises[specialize(item)] = process(specialize(item));
   }
 
-  while (running.size > done.size) {
-    await Promise.all(promises);
+  const abortSignal = AbortSignal.timeout(10 * 60 * 1_000);
```

Comment:
Can you assign a const to this time out so we know how much timeout it is? I think it's like 10 min?